lib/PrintAsObjC/PrintAsObjC.cpp

```diff
@@ -912,29 +911,42 @@ class ObjCPrinter : private DeclVisitor<ObjCPrinter>,
     }
     return hasPrintedAnything;
   }
-    
-  void printRenameForDecl(const AvailableAttr *AvAttr, const ValueDecl *D,
-                          bool includeQuotes) {
-    assert(!AvAttr->Rename.empty());
-    
-    auto renamedParsedDeclName = parseDeclName(AvAttr->Rename);
-    auto renamedDeclName = renamedParsedDeclName.formDeclName(D->getASTContext());
-    
+
+  const ValueDecl *getRenameDecl(const ValueDecl *D,
+                                 const ParsedDeclName renamedParsedDeclName) {
     auto declContext = D->getDeclContext();
-    const ValueDecl *renamedDecl = nullptr;
-    
+    ASTContext &astContext = D->getASTContext();
+    auto renamedDeclName = renamedParsedDeclName.formDeclName(astContext);
+
     if (isa<ClassDecl>(D) || isa<ProtocolDecl>(D)) {
+      if (!renamedParsedDeclName.ContextName.empty()) {
+        return nullptr;
+      }
       UnqualifiedLookup lookup(renamedDeclName.getBaseIdentifier(),
-                               declContext->getModuleScopeContext(),
-                               nullptr,
+                               declContext->getModuleScopeContext(), nullptr,
                                SourceLoc(),
                                UnqualifiedLookup::Flags::TypeLookup);
-      renamedDecl = lookup.getSingleTypeResult();
+      return lookup.getSingleTypeResult();
+    }
+    
+    TypeDecl *typeDecl = declContext->getSelfNominalTypeDecl();
+    
+    const ValueDecl *renamedDecl = nullptr;
+    SmallVector<ValueDecl *, 4> lookupResults;
+    declContext->lookupQualified(typeDecl, renamedDeclName,
```

Comment:
Oops, you probably want `declContext->getDeclaredInterfaceType` here.

Reply:
Oh thank you. I changed this line to `declContext->lookupQualified(typeDecl ->getDeclaredInterfaceType...` instead of `declContext->getDeclaredInterfaceType` as you suggested. Is that OK

Reply:
The difference is whether you take generic parameters into account in a constrained extension, but for Objective-C that shouldn't matter, since you can't expose a constrained extension method to Objective-C.